File: BackEnd/customersapiservice/src/main/java/com/bej/customersapiservice/domain/Customer.java
package com.bej.customersapiservice.domain;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
public class Customer {
    @Id
    private String customerId;
    private String customerName;
    private String customerEmail;
    private String customerPassword;
    private String customerProfilePic;
    private long customerPhone;
    private Address customerAddress;
    private List<String> customerFavRestaurants;
    private List<Object> customerFavDishes;

}

File: FrontEnd/DishDash/src/app/Model/customer.ts
import { address } from "./address"
import { favdishes } from "./favdishes"
import { favrestaurant } from "./favrestaurant"
import { FileHandle } from "./file-handle"

export class customer {
    customerId:string
    customerName?: string
    customerEmail?: string
    customerPassword?: string
    customerProfilePic?: FileHandle
    customerPhone?: number
    customerAddress?:address[]
    customerFavRestaurants?:string[]
    customerFavDishes?:favdishes[]
    customerCartId?:string
}
